### PR TITLE
Ezpz cu 369rqzv: Create Forgot Password Page

### PR DESCRIPTION
**Pull request by KevinLaCarrubba**
- Created forgot password page linked to `/forgot-password`
- The email field is hooked up to the useForm hook.
- Upon submitting an alert banner will show and clear the form.
- Combined regex.ts and regexValidators.ts .. Remamed to regex.ts
- Moved validations to be exported form /validations/forms.ts to be used on form validation files.
*****
***Side Tasks ?***
- Added in a loader for future use cause i saw it on IG and i liked it. 
- Fixed width of the forms to sit at `30rem` because overall i think they all looked better like this.
***Side note***
- I did some shit possibly .. idk we shall seee .

### CODE REVIEW

**Comment by arfusop** [comment]
Task linked: [CU-369rqzv Create the Forgot Password page](https://app.clickup.com/t/369rqzv)

**Review by arfusop — CHANGES_REQUESTED** [review]
Few changes, few general comments/questions

File: components/views/auth/ForgotPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    const [disableSubmit, setDisableSubmit] = useState<boolean>(true)
+    const [alert, setAlert] = useState<{ type: string; email: string }>({
+        type: '',
+        email: ''
+    })
+    const [showBanner, setShowBanner] = useState<boolean>(false)
+
+    const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        setAlert({
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/views/auth/ForgotPassword.tsx:39]: thats what i figured so i didnt too in depth .. also why i didnt even bother using the loader after i put the code in .

File: components/views/auth/ResetPassword.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const ResetPassword = () => {
                                 onChange={handleChange}
                                 minLength={8}
                                 title="Custom"
-                                message={true}
+                                showMessage={true}
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/views/auth/ResetPassword.tsx:73]: Clean.

File: pages/forgot-password.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import ForgotPassword from '../components/views/auth/ForgotPassword'
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply pages/forgot-password.tsx:1]: Fixed.

File: components/loader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import styles from './Loader.module.scss'
+
+const Loader = () => {
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/loader/index.tsx:3]: ty ty

File: components/views/auth/Login.tsx [hunk 5]
```diff
@@ -79,7 +78,7 @@ const Login = () => {
                             Submit
                         </button>
                         <div className={styles.option}>
-                            <Link href="/forgotpassword">
+                            <Link href="/forgot-password">
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/views/auth/Login.tsx:81]: done

File: components/views/auth/ForgotPassword.tsx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    email: {
+        test: (value: string) => VALID_EMAIL.test(value),
+        message: 'Must containt a valid email address (example@test.com)'
+    }
+}
+
+const ForgotPassword = () => {
+    const { form, handleChange, handleReset } = useForm(
+        INITIAL_STATE,
+        VALIDATIONS
```
[the remark this answers is not in the document]
Comment (KevinLaCarrubba) [reply components/views/auth/ForgotPassword.tsx:27]: Phantom Errors

**Review by arfusop — APPROVED** [review]

**Review by SamiFdev — APPROVED** [review]

**Pull request merged**